### DISCUSSION TITLE
How can I analyze shadow.log

### DISCUSSION THREAD

**Discussion by gmhXing** [shadow/shadow#1130] [Q&A] [answered]
![捕获](https://user-images.githubusercontent.com/24390389/84597735-94a00b00-ae98-11ea-879c-33609ed0ce88.PNG)
What information does each field in the picture represent (separated by commas)?
Please help me!
Thank you!

**Comment by robgjansen** [comment, marked as answer]
Sorry - this question got overlooked!

The information that appears in the shadow-heartbeat lines is not well documented. Some information is provided in the log file if you grep for "node-header" (think of this like the header row in a table):

```
00:00:00.069310 [thread-0] 00:00:00.000000000 [message] [testnode~11.0.0.1] [tracker.c:431] [_tracker_logNode] [shadow-heartbeat] [node-header] interval-seconds,recv-bytes,send-bytes,cpu-percent,delayed-count,avgdelay-milliseconds;inbound-localhost-counters;outbound-localhost-counters;inbound-remote-counters;outbound-remote-counters where counters are: packets-total,bytes-total,packets-control,bytes-control-header,packets-control-retrans,bytes-control-header-retrans,packets-data,bytes-data-header,bytes-data-payload,packets-data-retrans,bytes-data-header-retrans,bytes-data-payload-retrans
```

Otherwise, until we have better docs, you would have to [dig into the code here](https://github.com/shadow/shadow/blob/d2fc305e3c4e06005118b0c6d98c942e653c0d45/src/main/host/tracker.c#L431) to find out more.
